### DISCUSSION TITLE
GPL v3.0 有个buff有空的注意一下

### DISCUSSION THREAD

**Discussion by Groupguanfang** [Darock-Studio/Darock-Bili#85] [常规]
@WindowsMEMZ 

GPL3.0明文规定所有文件的文件头部得添加类似下面的注释，声明版权。
但是这个要求`不是强制性`的，所以看看你们怎么抉择吧。

一个小示例：

```typescript
/*
 * Copyright (C) 2024 Zero naily.cc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
```

类似这样的注释，`idea`家族的IDE中有一个版权设置选项，可以用`Velocity`模版来配置版权声明：

<img width="982" alt="image" src="https://github.com/Darock-Studio/Darock-Bili/assets/64218589/a5e2eb98-b798-4f60-9c7b-beca1023b6d5">

配置好之后可以一键生成版权到整个项目的所有文件的文件头部。

但是我不知道`XCode`有无此种功能，按理应该是有的。

这事我就不开issue了，😂discussion讨论一下就行

**Comment by WindowsMEMZ** [comment]
里面有实际代码的文件都是有文件头的，只不过不是完全按照这种格式。既然不是强制性的，那么就继续按照 Darock 的写法来写文件头注释吧。

至于一些文件没有版权注释，那是因为它们里面**完全没有**实际工作的代码，例如当前main下的`~/Bangumi/*.swift`，这些文件会在它们被写入实际代码时添加头注释。
